server/src/main/java/com/linecorp/centraldogma/server/internal/JGitUtil.java

```diff
@@ -0,0 +1,106 @@
+import static org.eclipse.jgit.lib.ConfigConstants.CONFIG_KEY_HIDEDOTFILES;
+import static org.eclipse.jgit.lib.ConfigConstants.CONFIG_KEY_LOGALLREFUPDATES;
+import static org.eclipse.jgit.lib.ConfigConstants.CONFIG_KEY_PRECOMPOSEUNICODE;
+import static org.eclipse.jgit.lib.ConfigConstants.CONFIG_KEY_RENAMES;
+import static org.eclipse.jgit.lib.ConfigConstants.CONFIG_KEY_REPO_FORMAT_VERSION;
+import static org.eclipse.jgit.lib.ConfigConstants.CONFIG_KEY_SYMLINKS;
+import static org.eclipse.jgit.lib.ConfigConstants.CONFIG_RECEIVE_SECTION;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.eclipse.jgit.lib.Config;
+import org.eclipse.jgit.lib.StoredConfig;
+
+public final class JGitUtil {
+
+    public static final int REPO_FORMAT_VERSION = 1;
+
+    private static final String REPO_FORMAT_VERSION_STR = String.valueOf(REPO_FORMAT_VERSION);
+
+    public static void applyDefaultsAndSave(StoredConfig config) throws IOException {
+        if (applyDefaults(config)) {
+            config.save();
+        }
+    }
+
+    public static boolean applyDefaults(Config config) {
+        boolean updated = false;
+        // Update the repository settings to upgrade to format version 1 and reftree.
+        updated |= set(config, CONFIG_CORE_SECTION, CONFIG_KEY_REPO_FORMAT_VERSION, REPO_FORMAT_VERSION_STR);
+
+        // Disable hidden files, symlinks and file modes we do not use.
+        updated |= set(config, CONFIG_CORE_SECTION, CONFIG_KEY_HIDEDOTFILES, "false");
+        updated |= set(config, CONFIG_CORE_SECTION, CONFIG_KEY_SYMLINKS, "false");
+        updated |= set(config, CONFIG_CORE_SECTION, CONFIG_KEY_FILEMODE, "false");
+
+        // Don't log ref updates.
+        updated |= set(config, CONFIG_CORE_SECTION, CONFIG_KEY_LOGALLREFUPDATES, "false");
+
+        // Do not decompose file names in macOS.
+        updated |= set(config, CONFIG_CORE_SECTION, CONFIG_KEY_PRECOMPOSEUNICODE, "true");
+
+        // Disable GPG signing.
+        updated |= set(config, CONFIG_COMMIT_SECTION, CONFIG_KEY_GPGSIGN, "false");
+
+        // Set the diff algorithm.
+        updated |= set(config, CONFIG_DIFF_SECTION, CONFIG_KEY_ALGORITHM, "histogram");
+
+        // Disable rename detection which we do not use.
+        updated |= set(config, CONFIG_DIFF_SECTION, CONFIG_KEY_RENAMES, "false");
+
+        // Disable auto-GC by default because of the following reasons:
+        // - GC can take long time affecting performance.
+        // - jGit's GC task has memory leak which adds a JVM shutdown hook for each GC task.
+        //   See https://github.com/eclipse-mirrors/org.eclipse.jgit/blob/2e3f12a0fc63deba80e725bfa985c0c5bd31de99/org.eclipse.jgit/src/org/eclipse/jgit/internal/storage/file/GC.java#L1769
+        updated |= set(config, CONFIG_GC_SECTION, CONFIG_KEY_AUTO, "0");
+        updated |= set(config, CONFIG_GC_SECTION, CONFIG_KEY_AUTOPACKLIMIT, "0");
+        updated |= set(config, CONFIG_RECEIVE_SECTION, CONFIG_KEY_AUTOGC, "false");
```

Comment:
```suggestion
```

Probably not too important, but I believe since `needGc()` is always false thanks to the settings above and `automatic == true` for auto gc anyways, this setting isn't needed. (there are other code paths calling `FileRepository#autoGC` anyways)

https://github.com/eclipse-mirrors/org.eclipse.jgit/blob/f1a9adf7da3cd130a0233aad3f31e2917317c13a/org.eclipse.jgit/src/org/eclipse/jgit/internal/storage/file/GC.java#L275

I'm fine with leaving this config if it makes sense semantically though

Reply:
Yeah, this setting is not really needed for this specific case. I'm just trying to be pre-cautious. Let me leave this as it is.